### ISSUE TITLE
Bungeecord pass event

### ISSUE THREAD

**Issue by GeorgeMixas** [BenCodez/VotingPlugin#661]
**Version**

5.20.3
Bungeecord snapshot:9e76966:1421
Paper 209

**Context**

Hello, I'm running NuVotifierr and VotingPlugin on my Bungeecord server and i was wondering if the Bungeecord version of VotingPlugin is able to understand vote events and pass them to other servers or if I have to use NuVotifier + VotingPlugin on a hub spigot server and pass the votes to the rest of the network from there.

Thanks.

**Comment by BenCodez** [comment]
You can do both, but running VotingPlugin on bungee is still sort of
experminetal. There is a wiki page detailing this more

**Comment by GeorgeMixas** [comment]
Okay so after some testing I can't get VotingPlugin to work on Bungeecord it doesn't have a debug mode so I don't really know if it gets the vote from NuVotifier but the MYSQL table stays empty after voting so it eather doesn't update the MySQL or it doesn't understand the vote from NuVotifier.

Spigot works though so I will just vote on my hub server and send the votes to the rest of the network from there.

Tell me if there is any way I can help you (logs, configs etc) so you can get the Bungeecord version of the plugin working.

**Comment by BenCodez** [comment]
I wouldn't worry about it, I'm changing how bungeecord is going to work in future versions

**Comment by GeorgeMixas** [comment]
I'm having a really hard time with this plugin for some reason 

it just doesn't want to communicate with the rest of the network

This is the setup that I'm using 
bungee: nothing
lobby: VotingPlugin, NuVotifier
survival: VotingPlugin

When I'm on Lobby and vote it works as expected but when I'm on survival and vote it doesn't work (Lobby gets the vote but it doesn't forward it to survival) 

Lobby config (https://pastebin.com/zq8QFvfG)
survival config (https://pastebin.com/KMdCHdZy)

I have tried all the DataStorage options.

I would really appreciate some help I have been trying many hours to get it to work but I just can't.

I'm using the latest version of paper, NuVotifier and Jenkins #469 for VotingPlugin.

**Comment by BenCodez** [comment]
It doesn't update instantly, rewards and votesites need to be the same across servers. Configs look good from what I can see

**Comment by GeorgeMixas** [comment]
Rewards and vote sites are the same, have been in survival for more than an hour and no votes 

also, should the MySQL table look this empty? (https://prnt.sc/ppf27p)

**Comment by BenCodez** [comment]
Looks like a mysql issue then. I don't know how to help you with that. Does it work with other plugins?

**Comment by GeorgeMixas** [comment]
Litebans works
I will try another database and see if that fixes it, Do you have any recommendations? like database providers that you have tested before and you know for a fact that are working?

**Comment by BenCodez** [comment]
No idea right now.

**Comment by GeorgeMixas** [comment]
I'm getting this in the console when I restart the server
https://pastebin.com/4NUBQGWx
tried 4 deferent databases all give the same error as well as the one original one.

**Comment by BenCodez** [comment]
Don't use the dev build right now

**Comment by GeorgeMixas** [comment]
Used 3 deferent databases with no errors in the console (5.20.3 no dev build) and i have the same problem MySQL tables will not update and the vote will not get to survival.

**Comment by GeorgeMixas** [comment]
Alright so after 6 databases I found one that works it was a database problem after all.

I will close this for now.

Thanks for the help!

**Comment by GeorgeMixas** [comment]
I will close this for now.

**Comment by BenCodez** [comment]
What database ended up working?

**Comment by GeorgeMixas** [comment]
https://www.freemysqlhosting.net/
Is a free one that ended up working
I will also provide a paid one when I find it

Also, I have two questions: 

1) Is there a way to make the votes between servers faster? it can take ~1-3 minutes for the votes to get to another server

2) I'm talking to the hosting provider that I use for the first database that did not work with the plugin, Do you have any idea why that database did not work? 
Any weird tricks your plugin does to manage the database and that's why it doesn't work with many providers or something like that?

**Comment by BenCodez** [comment]
It actually turns out that there is a bug in 5.20.3 which is probably the issue. 
This should fix it: http://ben12345rocks.com/job/VotingPlugin/

1. Set the background update to 1 minute. or directly go the server for now and change the new bungee method in the future?

**Comment by GeorgeMixas** [comment]
Tried the above build but it did not work this is what I got: 

https://pastebin.com/DdNZjFPm

**Comment by BenCodez** [comment]
Is bungeecord set to true?

**Comment by GeorgeMixas** [comment]
On both servers

**Comment by BenCodez** [comment]
They should be all false

**Comment by GeorgeMixas** [comment]
alright so it looks like the MySQL issue is resolved but I have another one, if I am on survival and vote on the server with the votifier (lobby) I will get about 1-2 votes on survival but after that no more votes will come to survival, I will have to relogin (join lobby and then survival) for any more votes to go trough

**Comment by BenCodez** [comment]
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

Try that. Don't know about the vote thing though. Clearcache options enabled on lobby?

**Comment by GeorgeMixas** [comment]
Nope its not should it be? 

Will also try that build and report back

**Comment by BenCodez** [comment]
Enable it, that would be the issue of the more than 1 vote I think.

**Comment by GeorgeMixas** [comment]
MySQL works in this build as well but the above problem still remains, let me rephrase it a little
I Join on survival (through the lobby) I vote once on survival the vote goes through I get my items and everything works, then I vote again on survival and the second vote doesn't go through basically every vote after the first one doesn't go to survival

**Comment by BenCodez** [comment]
Is that with the clear cache options on lobby? And just to confirm, same votesites files across servers?

**Comment by GeorgeMixas** [comment]
Which clearcache options should I enable on lobby there are 3

ClearCacheOnUpdate: false
ClearCacheOnJoin: false
ClearCacheOnVote: false

Also after some more testing it looks like only the first vote that enters the database works if I want another vote to work I will have to change this

//Table name prefix, for use with mutliple servers
Prefix: 'network2'

but even if I change it and create a new table again only the first vote on that new table works

and all the votesites are the same.

**Comment by BenCodez** [comment]
All of them.

**Comment by GeorgeMixas** [comment]
Did that and it did not help

**Comment by BenCodez** [comment]
What happens if you /av clearcache? Does it process the reward then?

**Comment by GeorgeMixas** [comment]
If I do that it always updates the total votes correctly but I don't always get the rewards for example if I'm on survival and vote on lobby there 3 thins that can happen after I vote (only once) and do /av clearcache 
1) I get nothing 
2) I get the rewards for the 1 vote (That's what should happen all the time)
3) I get more than 1 reward for the 1 vote

**Comment by BenCodez** [comment]
Hoping this helps:
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

Make sure all clear cache options (all 3) and always update, it's required for it to work.

I'm thinking it might be an issue in the later versions, as there is someone else if the same problem, so you could try downgrading a version or 2.

**Comment by GeorgeMixas** [comment]
I seriously have no idea what I'm doing wrong I tried using an older version with a deferent database as well but no luck please take a look at the configs just to make sure that everything is set up correctly because I'm running out of ideas on how to make the plugin work

Lobby Config:
https://pastebin.com/k1P1WXzS
Lobby VoteSites:
https://pastebin.com/wauQqBCW

Survival Config:
https://pastebin.com/KPnpappP
Survival VoteSites:
https://pastebin.com/8EE5fjNr

**Comment by BenCodez** [comment]
If your interested, you could try out the latest dev build and new way of
doing bungeecord, it's a work in progress, but I can help you with it
tomorrow

**Comment by GeorgeMixas** [comment]
Yea sure I would love to test it out I will try it and let you know how it went

**Comment by BenCodez** [comment]
There isn't much details on it, I can sort of walk you through it tomorrow

**Comment by BenCodez** [comment]
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

https://github.com/Ben12345rocks/VotingPlugin/wiki/Bungeecoord-Setups
First method on the wiki page details mostly everything, let me know if you need if you need more help than that.

VotingPlugin on bungeecoord with nuvotifier with sockets (like votifier) on each server. Check the config on github to see the new options

**Comment by GeorgeMixas** [comment]
Sorry for the late reply but I had some real life problems and I could not work on the server.
So basically bungeecord needs to have 3 ports right? one for bungee one for nuvotifier and one for votingplugin correct?

**Comment by GeorgeMixas** [comment]
and every other server needs 2 one for the server itself and one for votingplugin?

**Comment by BenCodez** [comment]
Port for nuvotifier, port for votingplugin on every server (with it configured in each config)

**Comment by giorgakism1** [comment]
So basically every server needs to have nuvotifier?

**Comment by BenCodez** [comment]
No, just bungee.

**Comment by GeorgeMixas** [comment]
Okay so this is the setup: 

Bungee: votingplugin, Nuvotifier
lobby: votingplugin

I have setup MySQL on bungee but after voting the database remains empty is that normal?

**Comment by BenCodez** [comment]
https://github.com/Ben12345rocks/VotingPlugin/wiki/Bungeecoord-Setups

You need to do everything for the first setup, make sure you use the latest version.

**Comment by GeorgeMixas** [comment]
I have done everything but the vote doesn't go thought and the MySQL doesn't update do you want to take a look at the configs?

**Comment by BenCodez** [comment]
try this command on bungee /votingpluginbungee vote (name) (site)

**Comment by GeorgeMixas** [comment]
Tried it did not work mysql did not update and the vote did not appear at lobby

**Comment by GeorgeMixas** [comment]
Do you maybe want to take a look at the configs?

**Comment by GeorgeMixas** [comment]
Actually I give up I will just forward with nuvotifier and have your plugin for storage I have wasted WAY to much time on this plugin you can close this if you want

**Comment by BenCodez** [comment]
Guess you could of just sent the configs and I could atleast figure out what's not working.

**Comment by BenCodez** [comment]
I also want to mention I figured out the issue with the databases you had before, fixed in the latest version.

**Comment by GeorgeMixas** [comment]
Yep, I know MySQL kinda works when using the spigot version of the plugin but when using the bungeecord version of the plugin it doesn't I will share the configs in a few hours when I have time

**Comment by BenCodez** [comment]
Bungee version works differently, doesn't directly save to mysql, it sends a message to the spigot servers to handle the vote. My guess is the message can't send.
